baseplate/lib/message_queue.py

```diff
@@ -103,22 +139,64 @@ def put(self, message: bytes, timeout: Optional[float] = None) -> None:
         raise TimedOutError
 
     def unlink(self) -> None:
-        """Remove the queue from the system.
+        self.queue.unlink()
 
-        The queue will not leave until the last active user closes it.
+    def close(self) -> None:
+        self.queue.close()
 
-        """
-        self.queue.unlink()
+
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue.""" # Used by the sidecar
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
+
+    def unlink(self) -> None:
+        """Not implemented for in-memory queue"""
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        """Not implemented for in-memory queue"""
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class RemoteMessageQueue(MessageQueue): 
+    def __init__(self, max_messages: int): # Connect to the sidecar and instantiate a remote queue
+        transport = TSocket.TSocket( "localhost" , 9090) # todo: how do I connect to the sidecar? I dont think this is right
+        transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(transport)
+
+        self.client = RemoteMessageQueueService.Client(protocol)
+        # todo: create queue?
+
+        # Connect to server
+        self.transport.open()
+
+    def get(self, timeout: Optional[float] = None) -> bytes: 
+        return self.client.get(timeout)
```

Comment:
What is the behaviour of this method if the queue is empty? You should ensure it matches the other classes which seem to raise `TimedOutError`s if the queue is empty.